Saulius merge all

sauliusg opened the pull request:
Merge data overview and analysis tables and Makefiles to regenerate them, with minimal comments.

Commit:
modified pre-commit for major python version 3

Commit:
added gitignore file

Commit:
Merge branch 'dev' into dev-markus

Commit:
change after merge

Commit:
added "correct" pytest coverage

Commit:
Modified developer README-dev to document makefile functionality

ppegolo committed:
Add Frontiers data with fixed inconsistencies

ppegolo committed:
Merge branch 'dev' into fix-frontiers

ppegolo committed:
Add IOP data

ppegolo committed:
Merge branch 'dev' into fix-frontiers

ppegolo committed:
Merge branch 'dev' into fix-IOP

ppegolo committed:
Update IOP data

ppegolo committed:
Merge branch 'dev' into fix-IOP

ppegolo committed:
Merge branch 'dev' into fix-frontiers

ppegolo committed:
Update data

pagramos committed:
Add files via upload

lukas-hoermann committed:
added my encodings to the rdp.csv file

lukas-hoermann committed:
Fixed data added to rdp.csv, added notebook to fix data

lukas-hoermann committed:
More fixes to rdp.csv

lukas-hoermann committed:
added notebooks to resolve conflicting question on APS and T&F

lukas-hoermann committed:
Fixed output of question resolution

lukas-hoermann committed:
Merge branch 'dev' into fix-aps-tf

lukas-hoermann committed:
Update file_handling.py

lukas-hoermann committed:
Update file_handling.py

lukas-hoermann committed:
Update file_handling.py

lukas-hoermann committed:
added corrected answers

lukas-hoermann committed:
update to file handeling

lukas-hoermann committed:
linting

ppegolo committed:
Merge branch 'dev' into fix-IOP

ppegolo committed:
Update files

ppegolo committed:
Merge branch 'dev' into fix-IOP

ppegolo committed:
Merge branch 'dev' into fix-IOP

ppegolo committed:
Clean up example

ppegolo committed:
Merge branch 'dev' into fix-frontiers

ppegolo committed:
Add discrepancy reason

lukas-hoermann committed:
Merge branch 'dev' into fix-aps-tf

Commit:
Commit prior to merge

Commit:
Merge remote-tracking branch 'origin/dev' into dev-markus

lukas-hoermann committed:
Pre commits run on file_handling

lukas-hoermann committed:
Merge branch 'fix-aps-tf' of github.com:daemoncost/Journal-Research-Data-Policy into fix-aps-tf

lukas-hoermann committed:
Second try running pre-commits and making some manual changes

Commit:
Merge branch 'fix-IOP' into saulius-merge-all

Commit:
Merge branch 'fix-aps-tf' into saulius-merge-all

Commit:
Merge branch 'fix-frontiers' into saulius-merge-all

Commit:
Merge branch 'fix-nature' into saulius-merge-all

Commit:
Merge branch 'fix-optica' into saulius-merge-all

Commit:
Adding two shell command histories to extract data from YAML files.

Commit:
Merge branch 'dev' into dev-markus

Commit:
Moving all history files to a subdirectory.

Commit:
Generating an overview of the collected answers in the TSV format.

Commit:
Generating the same overview in a more human-readable TXT format.

Commit:
Moving all my previous analysis files into ata/analysis/ subdirectory.

Commit:
Added `pretty` target for autoformatting python code

Commit:
Setting up the Makefiles to reproduce the computations.

Commit:
Generating the TXT overview using the Makefile.

Commit:
Generating the same data in TSV and CSV formats.

Commit:
Regenerating the files with corrected 'echo' statement for CSV headers.

Commit:
added linter in makefile

Commit:
Marking my TO-DO tasks.

Commit:
Merge branch 'dev-markus' into saulius-merge-all

Commit:
Merge branch 'dev' into saulius-merge-all – adding all new data and changes.

Commit:
Merge branch 'dev' into dev-markus

Commit:
Fixing two typos.

Commit:
Generating a 'sed' script from the Python input to change numbers to
keywords in the YAML files.

Commit:
Converting numeric keys to keywords using the keywird values extracted
from the *.py file.

Commit:
Regenerating all output data files. The only change is the order of the
answers.

Commit:
Adding the new keywords to the text overview.

Commit:
Regenerating also the TSV and CSV files. Leaving just one line with the
'topic' in the TSV data to avoid non-macthing strings and strings with
wrong number of columns.

Commit:
Adding all lines with the corresponding topic key. It seems that this
also fixes the wrong counts.

Commit:
Merge branch 'dev-markus' into saulius-merge-all

Commit:
Ignoring LibreOffice lock files.

Commit:
Copying one YAML file with all keys present as a template file to get
all keys from it (usingh the "yq 'keys[]' ..." command).

Commit:
Adding a Make rule to generate a TSV-like file with all answers.

Commit:
Updating the 'clean' target to avoid conflicts.

Commit:
Adding column names for all available answers, to avoid wrong sorting
with LibreOffice.

sauliusg committed:
Fixing a typo.

sauliusg committed:
Moving deletion of the asnwer overview files from 'clean' to 'distclean'
target.

sauliusg committed:
Adding the first rule to generate a table with selected correct answer.
Adding comments to the makefile.
Deleting initial "all-answers" files on reacing the 'distclean', not the
'clean' target.

sauliusg committed:
Addint the generated tables with the selected answer.

sauliusg committed:
Adding notes after reviewing the newly generated tables.

sauliusg committed:
Adding command history that shows how discrepancies can be checked.

sauliusg committed:
Marking the TO-DO list items.

ppegolo commented:
Hi Saulius, thanks for opening the PR. I've tried using your code, but I haven't succeded. The Makefile in `data/analysis/saulius-generate-overview` seems to have no effect. Could you please add a comment on how to run the scripts?
I also see conflicts between this branch and the current version of `dev`. Are the conflicting data in `dev` going to be substituted with other data in your branch?

sauliusg committed:
Spelling the file name in correct English.

sauliusg committed:
Changing "topics" to "topic" (singular) in the outputs of
"Makelocal-gen-overview". The "outputs/topic-answer-overview.txt" data
dod not change (only the timestamp).

sauliusg committed:
Generated templates for describing columns in the output TSV files.

sauliusg committed:
Adding documentation of the columns in "topic-answer-overview".

sauliusg committed:
Adding column descritions of the "doc/selected-answer_columns.tsv" file.

sauliusg committed:
Adding the descritions of columns in the CSV column descritions (aka
meta-metadata)

The descritpion was taken from:

Path: /home/saulius/src/csv-metadata
Working Copy Root Path: /home/saulius/src/csv-metadata
URL: svn+ssh://saulius.grazulis.lt/home/saulius/svn-repositories/csv-metadata/trunk
Relative URL: ^/trunk
Repository Root: svn+ssh://saulius.grazulis.lt/home/saulius/svn-repositories/csv-metadata
Repository UUID: 0dc92757-0ded-43eb-9f6b-4bc2b1f549c6
Revision: 5
Node Kind: directory
Schedule: normal
Last Changed Author: saulius
Last Changed Rev: 5
Last Changed Date: 2022-06-24 12:02:01 +0300 (Fri, 24 Jun 2022)

sauliusg committed:
Fixing a typo.

sauliusg committed:
Adding cilumn descriptions and describing file format peculiarities in
'doc/all-answers_columns.tsv'.

sauliusg committed:
Distributing related files into common subdirectories.

sauliusg committed:
Fixing a typo.

sauliusg committed:
Adding a README file.

sauliusg committed:
Adding another example of command history that show how to get the
statistic of the answers.

sauliusg commented:
> Hi Saulius, thanks for opening the PR. I've tried using your code, but I haven't succeded. The Makefile in `data/analysis/saulius-generate-overview` seems to have no effect. Could you please add a comment on how to run the scripts?

Sure! I'll describe the use of the Make susbystem in README – would that be OK?

> I also see conflicts between this branch and the current version of `dev`. Are the conflicting data in `dev` going to be substituted with other data in your branch?

Then conflicts are there since my branch uses "Question IDs" for identifying questions, which are the keys like "rdp_exist" or "data_citability", as opposed to the old-style numeric keys ("1", "2") with reliance on the description text to identify the question.

I would suggest that we accept these changes to QuestionIDs, unless that would break heavily someone's current code (mine is already adapted to use the IDs).

JonathanSchmidt1 commented:
@sauliusg the question_ids dont seem to cause any problems but somehow this branch seems to contain less data that's readable for my code than dev.  
I dont think that should be the case or am I missing something:

number of journals for each publisher
dev: {'AIP': 14, 'APS': 11, 'Bentham Science': 6, 'Frontiers': 7, 'IOP': 18, 'Optica': 9, 'RSC': 9, 'Springer Nature': 9, 'Taylor & Francis': 15}


number of journals for each publisher
this branch: {'APS': 11, 'Bentham Science': 6, 'Frontiers': 7, 'IOP': 9, 'Optica': 9, 'Springer Nature': 9, 'Taylor & Francis': 15})

JonathanSchmidt1 commented:
Ok I guess the data is only in dev and when you merge it should be there.

sauliusg committed:
Adding description of gow to use 'make', and the prerequisites.

sauliusg committed:
Checking and correcting the spelling.

sauliusg commented:
> Hi Saulius, thanks for opening the PR. I've tried using your code, but I haven't succeeded. The Makefile in `data/analysis/saulius-generate-overview` seems to have no effect. Could you please add a comment on how to run the scripts?

I have added the run instructions to `README.txt`, together with the dependency list. The dependency list is currently in human prose, since I am not sure which platforms or OSes you are using, but it can be "scriptified" when more precise requirements are known; for example we have 'dependencies/<osr-name-version>/install.sh' scripts for the OSes that we use in our group, and I cold provide such automatic dependency installation scripts as well.

sauliusg commented:
> @sauliusg the question_ids dont seem to cause any problems but somehow this branch seems to contain less data that's readable for my code than dev. I dont think that should be the case or am I missing something:
> 
> number of journals for each publisher dev: {'AIP': 14, 'APS': 11, 'Bentham Science': 6, 'Frontiers': 7, 'IOP': 18, 'Optica': 9, 'RSC': 9, 'Springer Nature': 9, 'Taylor & Francis': 15}
> 
> number of journals for each publisher this branch: {'APS': 11, 'Bentham Science': 6, 'Frontiers': 7, 'IOP': 9, 'Optica': 9, 'Springer Nature': 9, 'Taylor & Francis': 15})

Apparently I did not get the AIP and possibly some other data, but when we merge all data should be united. The Makefile would find all new files that are added; due to an unfortunate naming convention used in the repo (directory names contain spaces) we (I) will have to make symlinks to the new files, but that's one 'ln' command, not a big deal.

sauliusg commented:
> @sauliusg the question_ids dont seem to cause any problems 

So I suggest that I mark conflicts as resolved in favor of QuestionIDs, and we merge then. All data should be the processable (I'll take care to update the tables when they are in dev).

sauliusg committed:
Merge branch 'dev' into saulius-merge-all

sauliusg committed:
Changing question numbers to quesion identifiers by rerunning the
command recorded in 'find-sed-patch-yaml.history'.

sauliusg committed:
Adding links to the new files.

sauliusg committed:
Automatically regenerating the output tables with the new data from
'dev'.

JonathanSchmidt1 committed:
fix black formtting in publisher.py

JonathanSchmidt1 commented:
Nice everything still works, I only noticed that IOP still has the numbers and not question_ids
